### PR TITLE
Moved balance fetching to cosmos client

### LINKED ISSUE #657: Problem : Move all obsolete tendermint client calls to the new cosmos client - Croeseid4 compatibility

### PR DESCRIPTION
**Pull request by crypto-3dd**
Fixes #657

### CODE REVIEW

File: src/service/rpc/NodeRpcService.ts [hunk 1]
```diff
@@ -64,8 +65,11 @@ export class NodeRpcService implements INodeRpcService {
   }
 
   public async loadAccountBalance(address: string, assetDenom: string): Promise<string> {
-    const response = await this.tendermintClient.getBalance(address, assetDenom);
-    return response?.amount ?? '0';
+    const response = await this.cosmosClient.get<BalanceResponse>(
+      `/cosmos/bank/v1beta1/balances/${address}/${assetDenom}`,
+    );
+    const balanceData = response?.data;
```
Comment (cdc-Hitesh) [inline src/service/rpc/NodeRpcService.ts:71]: May be we can introduce `denom` equality check with our `assetDenom`, just to be safe.

File: src/service/rpc/NodeRpcService.ts [hunk 2]
```diff
@@ -64,8 +65,11 @@ export class NodeRpcService implements INodeRpcService {
   }
 
   public async loadAccountBalance(address: string, assetDenom: string): Promise<string> {
-    const response = await this.tendermintClient.getBalance(address, assetDenom);
-    return response?.amount ?? '0';
+    const response = await this.cosmosClient.get<BalanceResponse>(
+      `/cosmos/bank/v1beta1/balances/${address}/${assetDenom}`,
```
Comment (cdc-Hitesh) [inline src/service/rpc/NodeRpcService.ts:69]: Sorry for the basic question,
Does this       `/cosmos/bank/v1beta1/balances/${address}/${assetDenom}`, can fetch from any Cosmos compatible blockchain?

**Review by cdc-Hitesh — COMMENTED** [review]
Left a few doubts.

File: src/service/rpc/NodeRpcService.ts (hunk 1 above)
Comment (crypto-3dd) [reply src/service/rpc/NodeRpcService.ts:71]: You see, the data returns by the endpoint does already the filtering. In fact, the caller even specifies the `denom`, so technically there's no need for another `denom` check, it returns a single object and that would just be redundant.

https://testnet-croeseid-4.crypto.org:1317/cosmos/bank/v1beta1/balances/tcro1lnj8dc9qazmlhm5t8vw8vryxyzv09zgep2cz9v/basetcro

File: src/service/rpc/NodeRpcService.ts (hunk 2 above)
Comment (crypto-3dd) [reply src/service/rpc/NodeRpcService.ts:69]: It's a standard endpoint for all cosmos-based chains like CRO, what's the difference is the base URL. Cro base URL will be different than let's say ATOM/COSMOS or any other forks.

You can check all other methods in that class to see where the same pattern is being re-used.

File: src/service/rpc/NodeRpcService.ts (hunk 1 above)
Comment (cdc-Hitesh) [reply src/service/rpc/NodeRpcService.ts:71]: okay sounds cool! 

Marking it resolved

File: src/service/rpc/NodeRpcService.ts (hunk 2 above)
Comment (cdc-Hitesh) [reply src/service/rpc/NodeRpcService.ts:69]: SUper super.
That is what I wanted to confirm.

Resolving it.

**Review by cdc-Hitesh — APPROVED** [review]
I assume the base branch is intentionally `dev` .
Rest all good @crypto-3dd

**Comment by crypto-3dd** [comment]
> I assume the base branch is intentionally `dev` .
> Rest all good @crypto-3dd

Yup that's because this fix is not related to EVM (Cronos) integration

**Pull request merged**